### PR TITLE
`learning_rate` as theano shared variable in Adam

### PR DESCRIPTION
**Pull request by johnarevalo**
So it can be monitored or modified using extensions.

### CODE REVIEW

File: blocks/algorithms/__init__.py [hunk 1]
```diff
@@ -789,7 +789,7 @@ class Adam(StepRule):
     def __init__(self, learning_rate=0.002,
                  beta1=0.1, beta2=0.001, epsilon=1e-8,
                  decay_factor=(1 - 1e-8)):
-        self.learning_rate = learning_rate
+        self.learning_rate = shared_floatx(learning_rate, "learning_rate")
```
Comment (rizar) [inline blocks/algorithms/__init__.py]: I support this change, bet let's be consistence, either all hyperparameters in an algorithm should be a shared variables, or none. Could you please do the same for `beta1`, `beta2`, etc.

Comment (dwf) [inline blocks/algorithms/__init__.py]: :+1:

**Comment by johnarevalo** [comment]
@rizar, rebase done. It also updates `AdaGrad`

**Comment by rizar** [comment]
Looks good, thanks!

**Pull request merged**